bigquery/nulls.go

```diff
@@ -181,6 +196,12 @@ func (n *NullFloat64) UnmarshalJSON(b []byte) error {
 	n.Float64 = 0
 	if bytes.Equal(b, jsonNull) {
 		return nil
+	} else if bytes.HasPrefix(b, posInf) || bytes.HasPrefix(b, posInf2) {
```

Comment:
Hey @shollyman 

I agree that `HasPrefix` might lead to corner cases and makes sense to change that to equality checks instead :) will do that shortly.

I was also using `HasPrefix` for two other reasons:
1. When Go itself prints values for `Infinity`, `-Infinity` it does so using the short form `+Inf` and `-Inf` (which the BigQuery api accepts happily)
2. The source of some data I'm parsing was also using the short form that was produced by Go.

To avoid a heap of customization in my, and others, code base' I was hoping to keep the unmarshaling short form support.

What do you think about :
- Changing this to equality checks only, no `HasPrefix` to avoid any corned cases.
- Change to always Marshal the long form of these values `Infinity`, `-Infinity` and `NaN`
- Keep the Unmarshal support for the short form equality checks as well as the long form. Doing this will avoid people having to create a custom type to support the way Go itself, and other languages, output this kind of data.

I can't see any harm also supporting the short form **for unmarshaling purposes only**, thoughts?

Reply:
That seems reasonable.  It's definitely the case that BQ is following the robustness principle in terms of parsing the variation, so it's reasonable we include that when unmarshalling.

Reply:
Thanks will make the changes and ping you in the PR once done.